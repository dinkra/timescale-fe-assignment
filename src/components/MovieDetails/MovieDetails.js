import styled from 'styled-components';

// helpers
import formatDate from '../../helpers/formatDate';

// components
// import { Typography } from '../../components';
// const { H4, Body } = Typography;

// styles
const H4 = styled.h4`
  margin: 0;
  font-family: inherit;
  font-weight: 700;
  font-size: ${({ theme }) => theme.fontSizeLarge};

  color: ${({ theme }) => theme.black};
`;

const Body = styled.div`
  font-family: inherit;
  font-weight: ${({ bold }) => (bold ? 700 : 400)};
  font-size: ${({ theme }) => theme.fontSizeNormal};

  color: ${({ theme }) => theme.black};
`;

const StyledHeader = styled(H4)`
  margin-bottom: 16px;
`;

const StyledWrapper = styled.div`
  display: flex;
  column-gap: 22px;
`;

const StyledImg = styled.img`
  width: 50%;
  height: fit-content;
`;

const StyledDescription = styled.div``;

const StyledBody = styled(Body)`
  margin-bottom: 16px;
`;

const MovieDetails = ({ movie, imgPath }) => {
  console.log(movie);
  return (
    <div>
      <StyledHeader>{movie.title}</StyledHeader>
      <StyledWrapper>
        <StyledImg src={imgPath} />
        <StyledDescription>
          <StyledBody>
            <strong>Release date:</strong> {formatDate(movie.release_date)}
          </StyledBody>
          <StyledBody>{movie.overview}</StyledBody>
          <Body>
            <strong>{movie.vote_average}</strong> / 10 ({movie.vote_count} total votes)
          </Body>
        </StyledDescription>
      </StyledWrapper>
    </div>
  );
};

export default MovieDetails;
