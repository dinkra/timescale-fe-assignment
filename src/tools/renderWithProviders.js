import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

// override render method
export { customRender as render };
