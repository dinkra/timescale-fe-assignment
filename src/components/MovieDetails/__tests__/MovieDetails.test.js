import { render } from '../../../tools/renderWithProviders';

import MovieDetails from '../MovieDetails';

describe('MovieDetails', () => {
  const movieMock = {
    adult: false,
    id: 246655,
    original_language: 'en',
    overview:
      "After the re-emergence of the world's first mutant, world-destroyer Apocalypse, the X-Men must unite to defeat his extinction level plan.",
    poster_path: '/qttNmCib9gHhR5q0QoZ3FgmGom9.jpg',
    title: 'X-Men: Apocalypse',
    vote_average: 6.5
  };

  test('should match snapshot', () => {
    const result = render(<MovieDetails movie={movieMock} />);
    expect(result).toMatchSnapshot();
  });
});
