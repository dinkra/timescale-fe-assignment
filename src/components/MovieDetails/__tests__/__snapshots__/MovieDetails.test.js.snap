// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h4
          class="sc-gtsrHT sc-hKFxyN lazLpQ cZiHPc"
        >
          X-Men: Apocalypse
        </h4>
        <div
          class="sc-eCApnc daxkQE"
        >
          <img
            class="sc-jSFjdj cMbRHf"
          />
          <div>
            <div
              class="sc-dlnjwi sc-gKAaRy iagrlf hBugTN"
            >
              <strong>
                Release date:
              </strong>
               
              December 15, 2022
            </div>
            <div
              class="sc-dlnjwi sc-gKAaRy iagrlf hBugTN"
            >
              After the re-emergence of the world's first mutant, world-destroyer Apocalypse, the X-Men must unite to defeat his extinction level plan.
            </div>
            <div
              class="sc-dlnjwi iagrlf"
            >
              <strong>
                6.5
              </strong>
               / 10 (
               total votes)
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h4
        class="sc-gtsrHT sc-hKFxyN lazLpQ cZiHPc"
      >
        X-Men: Apocalypse
      </h4>
      <div
        class="sc-eCApnc daxkQE"
      >
        <img
          class="sc-jSFjdj cMbRHf"
        />
        <div>
          <div
            class="sc-dlnjwi sc-gKAaRy iagrlf hBugTN"
          >
            <strong>
              Release date:
            </strong>
             
            December 15, 2022
          </div>
          <div
            class="sc-dlnjwi sc-gKAaRy iagrlf hBugTN"
          >
            After the re-emergence of the world's first mutant, world-destroyer Apocalypse, the X-Men must unite to defeat his extinction level plan.
          </div>
          <div
            class="sc-dlnjwi iagrlf"
          >
            <strong>
              6.5
            </strong>
             / 10 (
             total votes)
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
